ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/PartitionEvaluator.java

```diff
@@ -23,16 +23,28 @@
 import com.netease.arctic.server.optimizing.OptimizingType;
 
 import java.util.List;
+import java.util.Map;
 
 public interface PartitionEvaluator {
 
+  /**
+   * Weight determines the priority of partition execution, with higher weights having higher priority.
+   */
+  interface Weight extends Comparable<Weight> {
+
+  }
+
   String getPartition();
 
   void addFile(IcebergDataFile dataFile, List<IcebergContentFile<?>> deletes);
+  
+  void addPartitionProperties(Map<String, String> properties);
 
   boolean isNecessary();
 
   long getCost();
+  
+  Weight getWeight();
```

Comment:
Can `getWeight` override the usage of `getCost`?

Reply:
`getCost` cannot be replaced by `getWeight` because `getCost` is also used to evaluate how many tasks can be executed at most in one optimizing.

But perhaps we can include `getCost` as part of `Weight`, like
```
  interface Weight extends Comparable<Weight> {
    long getCost();
  }
```